### PR TITLE
Media: ensure mime types list is an array

### PR DESCRIPTION
**Pull request by spacedmonkey**
## Summary

Follow up from https://github.com/google/web-stories-wp/pull/8067. If a user select a incompatible file type and crops it. Ensure an error is shown.  

### QA

- [x] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1. Open editor. 
2. Select document tab
3. Open media picker to select poster image. 
4. Drag .bmp file into media dialog. 
5. Click Select. 
6. See snackbar message. 

### UAT

- [x] UAT should use the same steps as above.

This PR can be tested by following these steps:

1.

## Checklist

- [ ] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [ ] I have tested this code to the best of my abilities
- [ ] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [ ] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [ ] This PR contains automated tests (unit, integration, and/or e2e) to verify the code works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [ ] I have added documentation where necessary
- [ ] I have added a matching `Type: XYZ` label to the PR

---

Fixes #

### CODE REVIEW

File: includes/Traits/Types.php [hunk 1]
```diff
@@ -151,7 +151,7 @@ public function get_allowed_image_mime_types() {
 		 */
 		$image_mime_type = apply_filters( 'web_stories_allowed_image_mime_types', $image_mime_type, $allowed_mime_type );
 
-		return array_intersect( $allowed_mime_type, $image_mime_type );
+		return array_values( array_intersect( $allowed_mime_type, $image_mime_type ) );
```
Comment (spacedmonkey) [inline includes/Traits/Types.php:154]: This brings this inline this line. 

https://github.com/google/web-stories-wp/blob/82088c76714904e49b9a289f45184b194316fe2d/includes/Traits/Types.php#L120

**Comment by swissspidy** [comment]
> If a user select an incompatible file type and crops it. Ensure an error is shown.

Why can an incompatible file type be selected in the first place? 🤔 

Could you add some testing instructions so I can better understand this?

**Comment by spacedmonkey** [comment]
> Could you add some testing instructions so I can better understand this?

Added QA details

**Comment by swissspidy** [comment]
> This PR can be tested by following these steps:
> 
> 1. Open editor.
> 2. Select document tab
> 3. Open media picker to select poster image.
> 4. Drag .bmp file into media dialog.
> 5. Click Select.
> 6. See snackbar message.

@spacedmonkey I can't really do that 🤔 

What happens for me on `main`:

1. Open editor
2. Select document tab
3. Open media picker to select poster image
4. Drag .bmp file into media dialog
5. Click "Select as poster image"
6. Get snackbar message `Please choose only gif, jpe, jpeg, jpg, png, or webp as a poster.`

**The same thing happens on this PR**, so I don't see a difference.

**Comment by spacedmonkey** [comment]
> The same thing happens on this PR, so I don't see a difference.

Yeah, I guess I was trying to overly careful here. The php change is still valid. Without it, if you filter the images types, it can end up as an object. 

Screenshot 2021-07-08 at 15 27 06

**Review by swissspidy — APPROVED** [review]

**Pull request merged**